### PR TITLE
prepare release v0.3.0

### PR DESCRIPTION
**Pull request by mengxr**
Prepare release 0.3:
* update README, versions
* update API doc in CC
* hide protected methods in `Logging`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,5 +30,9 @@ We welcome open source contributions as well!
 
 ## Releases:
 
- - 0.1.0 initial release
- - 0.2.0 release for Spark 2.0 (work of @felixcheung)
+- 0.1.0 initial release
+- 0.2.0 release for Spark 2.0 (work of @felixcheung)
+- 0.3.0
+  - DataFrame-based connected components implementation
+  - added support for Python 3
```
Comment (nchammas) [inline README.md]: I see there was a PR to address this in #135, but isn't this also dependent on https://github.com/databricks/sbt-spark-package/issues/26?

Comment (mengxr) [inline README.md]: Good point. Maybe we are not ready for announce it for this release.

**Comment by thunterdb** [comment]
LGTM

**Comment by nchammas** [comment]
Looks like GraphFrames 0.3 does indeed work with Python 3. I just tested it out successfully. Apologies about talking you out of including that in the release notes.

Shall I open a PR to re-add that into the release notes, or would one of you like to do it?

**Pull request merged**